File: dev-framework/resources/templates/projects/chat-interface/backend/app/api/models/sessions.py
#!/usr/bin/env python3
# app/api/models/sessions.py

from datetime import datetime
from typing import Optional
from uuid import UUID

from app.api.models.common import BaseSchema


# =====================================================
# REQUEST MODELS
# =====================================================

class SessionStartRequest(BaseSchema):
    prompt: str
    allowed_tools: list[str] | None = None


class SessionSendRequest(BaseSchema):
    prompt: str


# =====================================================
# RESPONSE MODELS
# =====================================================

class SessionResponse(BaseSchema):
    id: UUID
    claude_session_id: str | None = None
    status: str
    allowed_tools: str | None = None  # CSV string as-is (e.g. "Read,Write,Bash")
    workspace_id: UUID | None = None
    workspace_name: str | None = None
    workspace_color: str | None = None
    initiated_by: str = "human"
    first_message: str | None = None
    created_at: datetime
    updated_at: datetime


class MessageResponse(BaseSchema):
    id: UUID
    session_id: UUID
    role: str
    content: dict
    sequence_number: int
    created_at: datetime


class SessionHistoryResponse(BaseSchema):
    session: SessionResponse
    messages: list[MessageResponse]


class SessionListResponse(BaseSchema):
    sessions: list[SessionResponse]
    total: int

File: dev-framework/resources/templates/projects/chat-interface/frontend/src/types/sessions.ts
export type SessionStatus = 'active' | 'completed' | 'stopped' | 'error'

export interface Session {
  id: string
  claudeSessionId: string | null
  status: SessionStatus
  allowedTools: string | null
  workspaceId: string | null
  workspaceName: string | null
  workspaceColor: string | null
  initiatedBy: string
  firstMessage: string | null
  createdAt: string
  updatedAt: string
}

export interface FileAttachment {
  file_id: string
  filename: string
  mime_type: string
  size: number
}

export interface MessageContent {
  type: string
  result?: string
  attachments?: FileAttachment[]
  [key: string]: unknown
}

export interface Message {
  id: string
  sessionId: string
  role: 'user' | 'assistant'
  content: MessageContent
  sequenceNumber: number
  createdAt: string
}

export interface SessionHistory {
  session: Session
  messages: Message[]
}

export interface SessionListResponse {
  sessions: Session[]
  total: number
}

// =====================================================
// SSE Stream Event Types
// =====================================================

export interface TextBlock {
  type: 'text'
  text: string
}

export interface ToolUseBlock {
  type: 'tool_use'
  id: string
  name: string
  input: Record<string, unknown>
  output?: string
  is_error?: boolean
}

export type ContentBlock = TextBlock | ToolUseBlock

export interface AssistantStreamEvent {
  type: 'assistant'
  blocks: ContentBlock[]
}

export interface ResultStreamEvent {
  type: 'result'
  subtype: string
  result: string
}

export interface StatusStreamEvent {
  type: 'completed' | 'stopped' | 'error'
  message?: string
}

export interface ToolResultStreamEvent {
  type: 'tool_result'
  tool_use_id: string
  content: string
  is_error: boolean
}

export type StreamEvent = AssistantStreamEvent | ResultStreamEvent | StatusStreamEvent | ToolResultStreamEvent
